scripts/zones/Navukgo_Execution_Chamber/bcnms/omens.lua

```diff
@@ -0,0 +1,46 @@
+-----------------------------------
+-- Area: Navukgo Execution Chamber
+-- BCNM: BLU-AF2 Omens
+-----------------------------------
+require("scripts/globals/battlefield")
+require("scripts/globals/missions")
+local ID = require("scripts/zones/Navukgo_Execution_Chamber/IDs")
+----------------------------------------
+
+function onBattlefieldInitialise(battlefield)
+end
+
+function onBattlefieldTick(battlefield, tick)
+    dsp.battlefield.onBattlefieldTick(battlefield, tick)
+end
+
+function onBattlefieldRegister(player, battlefield)
+end
+
+function onBattlefieldEnter(player, battlefield)
+    local mobOffset = (battlefield:getArea() - 1) * 7  -- offset to spawn correct mob depending on battlefieldNumber
+    local players = battlefield:getPlayers()
+    if #players ==  4 then GetMobByID(17039375 + mobOffset):spawn(); end
+    if #players ==  8 then GetMobByID(17039376 + mobOffset):spawn(); end
+    if #players == 12 then GetMobByID(17039377 + mobOffset):spawn(); end
+    if #players == 14 then GetMobByID(17039378 + mobOffset):spawn(); end
+    if #players == 16 then GetMobByID(17039379 + mobOffset):spawn(); end
```

Comment:
Entity ID's should not be in anything other than `ID.lua`.

Also, I feel like the conditions you're checking are incorrect the way they are. If you're in a party/alliance of 5, 6, 7, 9, 10, 11, 13, 15 no additional mobs will spawn?

Reply:
onBattlefieldEnter gets called once per person as they enter, so the second mob gets spawned when the 4th person enters, nothing happens when player 5, 6, 7 enter, then the third mob spawns when player 8 enters. This would be a problem if it were in onBattlefieldInitialise, which just gets called only once when the battlefield is created.

Reply:
@cocosolos the way its written tho makes for this question to keep popping up every time a new person comes across it. I think the logic could be made to be clearer about whats happening so that future editors don't try to "fix" it.

Reply:
Made changes so it doesn't go through every if statement. Thank you for your input 🐱